examples/microprofile/oci-tls-certificates/README.md

```diff
@@ -0,0 +1,112 @@
+    4. Enter CN(Common Name), for example `my.super.authority`
+    5. Select your vault and the key `mySuperCAKey`
+    6. Select max validity for signed certs(or leave the default 90 days)
+    7. Check `Skip Revocation` to keep it simple
+    8. Select `Create Certificate Authority` button on the summary page
+    9. Notice OCID of the newly created CA, we will need it later
+
+### Configuration
+Following env variables to be configured in [config.sh](etc/unsupported-cert-tools/config.sh)
+for both [rotating](#rotating-mtls-certificates) certificates and [running](#build--run) the examples.
+
+- **COMPARTMENT_OCID** - OCID of compartment the services are in
+- **VAULT_CRYPTO_ENDPOINT** - Each OCI Vault has public crypto and management endpoints, we need to specify crypto endpoint of the vault we are rotating the private keys in (example expects both client and server to store private key in the same vault)
+- **VAULT_MANAGEMENT_ENDPOINT** - crypto endpoint of the vault we are rotating the private keys in
+- **CA_OCID** - OCID of the CA authority we have created in [Prepare CA](#prepare-cacertification-authority) step
+
+Following env variables are generated automatically by [createKeys.sh](etc/unsupported-cert-tools/createKeys.sh) or needs to be configured manually for [rotateKeys.sh](etc/unsupported-cert-tools/rotateKeys.sh) in [generated-config.sh](etc/unsupported-cert-tools/generated-config.sh)
+- **SERVER_CERT_OCID** - OCID of the server certificate(not the specific version!)
+- **SERVER_KEY_OCID** - OCID of the server private key in vault(not the specific version!)
+
+Optional:
+- **CLIENT_CERT_OCID** - OCID of the client certificate(not the specific version!)
+- **CLIENT_KEY_OCID** - OCID of the client private key in vault(not the specific version!)
+
+### Prepare keys and certificates
+Make sure you are in the directory [./etc/unsupported-cert-tools/](etc/unsupported-cert-tools/).
+```shell
+bash createKeys.sh
+```
+
+## Rotating mTLS certificates
+Make sure you are in the directory [./etc/unsupported-cert-tools/](etc/unsupported-cert-tools/).
+```shell
+bash rotateKeys.sh
+```
+⚠️ Keep in mind that rotation creates new [versions](https://docs.oracle.com/en-us/iaas/Content/certificates/rotation-states.htm), OCIDs of the keys and certificates stays the same, and you don't need to change your configuration.
+
+## Build and run example
+
+Update the [pom.xml](../pom.xml) to define the system properties for the configuration as mentioned above.
+
+### Build & Run
+
+```shell
+mvn clean package
+```
+
+Run mTLS secured web server:
+```shell
+source ./etc/unsupported-cert-tools/config.sh && \
+source ./etc/unsupported-cert-tools/generated-config.sh && \
+java -jar ./target/helidon-examples-microprofile-oci-tls-certificates.jar
+```
+
+### Test with cURL
+```shell
+curl --key key-pair.pem --cert cert-chain.cer --cacert ca.cer -v https://localhost:8443
+```
```

Comment:
We should explain how to obtain curl client's `key-pair.pem` `cert-chain.cer` and `ca.cer` files

Reply:
The example does not actually use or demonstrate the client side.

Reply:
But the curl example in the readme does

Reply:
IMHO we probably should actually show the java client usage, mTLS is very common question in our slack channel

Reply:
Filed https://github.com/helidon-io/helidon/issues/7507 as a follow-up issue.